app/assets/javascripts/galleries/add_new.js

```diff
@@ -52,6 +47,18 @@ function processDirectionalKey(event) {
   if (consume) event.preventDefault();
 }
 
+function processKeyLeft(caret) {
+  if (caret !== 0) { return false; }
+  $(this).closest('td').prev().find('input').focus();
+  return true;
+}
+
+function processKeyRight(caret, length) {
+  if (caret <= length) { return false; }
```

Comment:
Think this should be `caret < length` based on the previous condition being `caret >= length`